<!DOCTYPE html>
<html><head>

    <meta charset="utf-8">
	<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

    <link rel="canonical" href="https://synbiohub.github.io/userdocumentation/">

    <title>
        
        User Documentation | SynBioHub-Documentation
        
    </title>

    
    <link href="https://synbiohub.github.iocss/fontawesome.min.css" rel="stylesheet">

    
    <link rel="stylesheet" href="https://synbiohub.github.io/css/ace.min.css">

    
    
        
    

</head>
<body><nav class="navbar navbar-expand-lg navbar-dark bg-primary shadow sticky-top" id="navbarMain">
    <div class="container">
        <div>
            <a class="navbar-brand" href="https://synbiohub.github.io/">
                SynBioHub-Documentation
            </a>
        </div>
        
    </div>
</nav>
<div class="container-fluid">
            <div class="row">

                <div class="docs-sidenav order-0 col-12 col-md-3 col-lg-2 col-xl-2 position-sticky border-right"><nav class="navbar navbar-expand-md navbar-light pl-0">
    <button class="navbar-toggler navbar-toggler-right collapsed" type="button" data-toggle="collapse" data-target="#sidenav-left-collapse" aria-controls="sidenav-left-collapse" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

     <div class="collapse navbar-collapse align-items-start flex-column" id="sidenav-left-collapse">
            <form class="form-inline my-2 my-lg-0 searchbox">
                <input class="form-control mr-sm-2 w-100" data-search-input id="search-by" type="text" placeholder="Search">
            </form>
         <ul class="navbar-nav flex-column pt-3">
    <li data-nav-id="/aboutsynbiohub/" class="nav-item my-1
        ">
        
        <a class="nav-link p-0" href="https://synbiohub.github.io/aboutsynbiohub/"><h6>About SynBioHub</h6></a>
    </li>
    <li data-nav-id="/installation/" class="nav-item my-1
        ">
        
        <a class="nav-link p-0" href="https://synbiohub.github.io/installation/"><h6>Installation</h6></a>
    </li>
    <li data-nav-id="/setup/" class="nav-item my-1
        ">
        
        <a class="nav-link p-0" href="https://synbiohub.github.io/setup/"><h6>Setup</h6></a>
    </li>
    <li data-nav-id="/userdocumentation/" class="nav-item my-1 parent active
        ">
        
        <a class="nav-link p-0" href="https://synbiohub.github.io/userdocumentation/"><h6>User Documentation</h6></a>
    </li>
    <li data-nav-id="/apidocumentation/" class="nav-item my-1
        ">
        
        <a class="nav-link p-0" href="https://synbiohub.github.io/apidocumentation/"><h6>API Documentation</h6></a>
    </li>
    <li data-nav-id="/plugins/" class="nav-item my-1
        ">
        
        <a class="nav-link p-0" href="https://synbiohub.github.io/plugins/"><h6>Plugins</h6></a>
    </li>
    <li data-nav-id="/troubleshooting/" class="nav-item my-1
        ">
        
        <a class="nav-link p-0" href="https://synbiohub.github.io/troubleshooting/"><h6>Troubleshooting </h6></a>
    </li>
    <li data-nav-id="/references/" class="nav-item my-1
        ">
        
        <a class="nav-link p-0" href="https://synbiohub.github.io/references/"><h6> References</h6></a>
    </li>
        </ul>
    </div>
</nav>


</div>
                <div class="docs-toc large order-lg-2 order-md-0 order-xs-1 col-12 col-lg-2 col-xl-2 position-sticky border-left"><div class="docs-toc">
	<nav id="TableOfContents">
  <ul>
    <li><a href="#1-searching-for-information">1. Searching for Information</a>
      <ul>
        <li><a href="#11-searching-using-keywords">1.1 Searching using keywords</a></li>
        <li><a href="#12-sequence-search">1.2 Sequence Search</a></li>
        <li><a href="#13-advanced-search">1.3 Advanced Search</a></li>
        <li><a href="#14-sparql-search">1.4 SPARQL Search</a></li>
      </ul>
    </li>
    <li><a href="#2-registering--updating-an-account">2. Registering &amp; Updating an account</a>
      <ul>
        <li><a href="#21-registering-an-account">2.1 Registering an account</a></li>
        <li><a href="#22-updating-an-account">2.2 Updating an Account</a></li>
      </ul>
    </li>
    <li><a href="#3-viewing--downloading-the-information">3. Viewing &amp; Downloading the Information</a>
      <ul>
        <li><a href="#31-viewing-the-information">3.1 Viewing the Information</a></li>
        <li><a href="#32-downloading-the-information">3.2 Downloading the Information</a></li>
      </ul>
    </li>
    <li><a href="#4-submitting-managing--updating-submissions">4. Submitting, Managing &amp; Updating Submissions</a>
      <ul>
        <li><a href="#41-submitting">4.1 Submitting</a></li>
        <li><a href="#42-managing">4.2 Managing</a></li>
        <li><a href="#43-updating">4.3 Updating</a></li>
      </ul>
    </li>
    <li><a href="#5-editing-a-submission">5. Editing a Submission</a></li>
    <li><a href="#6-data-sharing-mechanisms">6. Data Sharing Mechanisms</a>
      <ul>
        <li><a href="#61-using-the-share-option">6.1 Using the Share option</a></li>
        <li><a href="#62-adding-an-owner">6.2 Adding an owner</a></li>
      </ul>
    </li>
    <li><a href="#7-administration">7. Administration</a>
      <ul>
        <li></li>
      </ul>
    </li>
  </ul>
</nav>
</div>
</div>
                <div class="main col-12 order-1 col-md-9 col-lg-10 col-xl-8 py-3">
                

<h1>User Documentation</h1>

<p>This section contains instructions on how you can implement various operations ranging from registering onto SynBioHub to uploading your genetic designs on SynBioHub, by using the user interface of SynBiohub. The sub-sections are as follows:</p>
<h2 id="1-searching-for-information">1. Searching for Information</h2>
<h3 id="11-searching-using-keywords">1.1 Searching using keywords</h3>
<p>To search amongst various resources, follow these steps:</p>
<ol>
<li>
<p>Navigate to SynBioHub&rsquo;s home page.</p>
</li>
<li>
<p>In the search box, enter the keywords for the resource you want to search for and click on the search button.</p>
</li>
<li>
<p>Subsequently, a list of resources matching the keywords that you&rsquo;ve entered shall appear on your screen.</p>
</li>
<li>
<p>Select the desired resource from the list.</p>
</li>
</ol>
<h3 id="12-sequence-search">1.2 Sequence Search</h3>
<p>To search using a sequence, follow these steps:</p>
<ol>
<li>
<p>Navigate to SynBioHub&rsquo;s home page.</p>
</li>
<li>
<p>Click on the <em>search</em> option. It&rsquo;ll direct you to a page containing a list of resources.</p>
</li>
<li>
<p>Located just below the search box are the various means through which you can search for the desired resources. In this case, click on <em>sequence search</em>.</p>
</li>
<li>
<p>Enter the sequence or upload a <em>FASTA/FASTQ</em> file, by clicking the choose file option just below the box. This shall search only for exact matches of the sequence. Your entry must be the first key/value pair. To know more about about FASTA files, click <a href="https://en.wikipedia.org/wikiFASTA_format#:~:text=In%20bioinformatics%20and%20biochemistry%2C%20the,comments%20to%20precede%20the%20sequences">here</a>.</p>
</li>
<li>
<p>Then, you can choose the perspective of your search. You can perform a <strong>global</strong> search or an <strong>exact</strong> search. The exact search shall display, those resources which are really close to the sequences that you&rsquo;ve entered and the global search search will display those results which have even the smallest of the similarities to the sequence that you&rsquo;ve entered.</p>
</li>
<li>
<p>Subsequently, you may customise the number of results, the value for which must not exceed 10,000. Then you need to enter the minimum and the maximum length of the sequence for which you want to search for. Values for both of these parameters should not exceed 100,000.</p>
</li>
<li>
<p>Then, you need to enter the maximum number of non-matching target sequences, that are to be considered before the search is halted. This option works in pair with the flag above. The search process sorts the target sequences by decreasing number of k-mers they have in common with the query sequence, using that information as a proxy for sequence similarity. After pairwise alignments, if none of the first x examined target sequences pass the acceptation criteria, the search process stops for that query (no hit). If this flag is set to a higher value, more target sequences are considered. If the flag above and this flag are both set to 0, the complete database is searched.</p>
</li>
<li>
<p>Fill up the <em>percentage match</em>, by entering a value between 0 &amp; 1. This depicts the percentage that you want the result to match your query.</p>
</li>
<li>
<p>Finally, select the <em>pairwise identity definition</em>. You may select the value for this attribute from the following options:</p>
</li>
</ol>
<table>
<thead>
<tr>
<th>Value</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>Default</td>
<td>Edits distance excluding the terminal gaps</td>
</tr>
<tr>
<td>CD-HIT definition</td>
<td>Considers the following ratio: (matching columns) / (shortest sequence length)</td>
</tr>
<tr>
<td>Marine Biological Lab definition</td>
<td>counts each gap opening (internal or terminal) as a single mismatch, whether or not the gap was extended: 1.0 - [(mismatches + gap openings)/(longest sequence length)]</td>
</tr>
<tr>
<td>BLAST Definition</td>
<td>equivalent to edit distance for global pairwise alignments</td>
</tr>
</tbody>
</table>
<h3 id="13-advanced-search">1.3 Advanced Search</h3>
<p>To perform an Advanced Search, follow these steps:</p>
<ol>
<li>
<p>Navigate to SynBioHub&rsquo;s home page.</p>
</li>
<li>
<p>Click on the <em>search</em> option. It&rsquo;ll direct you to a page containing a list of resources.</p>
</li>
<li>
<p>Located just below the search box are the various means through which you can search for the desired resources. In this case, click on <em>advanced search</em>.</p>
</li>
<li>
<p>Now, select the type of the resource you want to search for i.e, a collection, design, activity or a sequence.</p>
</li>
<li>
<p>Then, select the username of the design&rsquo;s creator.</p>
</li>
<li>
<p>Enter the timeframe between which the resource was created, by entering the <em>created after</em> and <em>created before</em> attributes. Make sure to enter the dates in the <strong>YYYY-MM-DD</strong> format.</p>
</li>
<li>
<p>If the resource has been modified, make sure to enter the <em>modified after</em> and <em>modified before</em> attributes as well. Make sure to enter the dates in the <strong>YYYY-MM-DD</strong> format.</p>
</li>
<li>
<p>Enter the ID or the name or the description of the resource, which you want to search for.</p>
</li>
<li>
<p>Select the <em>type</em> of the resource, from the following available options in the dropdown menu:</p>
</li>
</ol>
<ul>
<li>
<p>No filter</p>
</li>
<li>
<p>biopax:Complex</p>
</li>
<li>
<p>biopax:DnaRegion</p>
</li>
<li>
<p>biopax:Protein</p>
</li>
<li>
<p>so:circular</p>
<p>Subsequently, select the <em>role</em> of the resource from this dropdown menu. It contains various sub-categories of igem or 	SO.
To learn more about <strong>igem</strong>, visit <a href="https://igem.org/Main_Page">this page</a> and for <strong>SO</strong>, visit <a href="http://www.sequenceontology.org/miso/current_release/term/SO:0000316">this page</a></p>
</li>
</ul>
<ol start="10">
<li>
<p>You may select any number of collections, based on your convenience. To select a collection, click on the value attribute of collection and select it from the drop down menu.</p>
</li>
<li>
<p>You may apply customised filters for the design you want to search for. For that purpose, a few blank filtering columns have been provided.
As soon as you&rsquo;ll click on an empty filter, a drop down menu shall appear having various parameters for filtering. Select your preference, and then click on filter. This shall reload the page and then select the value of the filter, from the drop down on the right hand side. You may provide as many as 5 customised filters which&rsquo;er optional.</p>
</li>
</ol>
<h3 id="14-sparql-search">1.4 SPARQL Search</h3>
<p>To perform a SPARQL Search, the following steps should be followed:</p>
<ol>
<li>
<p>Navigate to SynBioHub&rsquo;s SPARQL Search Page.</p>
</li>
<li>
<p>Since, SPARQL is a query language, hence it&rsquo;ll query Synbiohub&rsquo;s databases for the designs that you&rsquo;re looking for.</p>
</li>
<li>
<p>Following is an example how a sample SPARQL query is written:</p>
</li>
</ol>
<pre><code>  SELECT  # makes up the headers of the table
	
        ?def # directs you to the design for which you're looking for 
	
        ?displayId # This statement displays the Id the design
	
        ?title # displays the title of the design
	
        (count(?def) as ?count) # displays the count parameter for the design
	
        ?role # displays the role of the design    

</code></pre><ol start="4">
<li>Next, you need to provide a <strong>Where</strong> statement, which shall contain all the specific details with respect to the <strong>SELECT</strong> statement. The Syntax is as follows:</li>
</ol>
<pre><code>WHERE { 
  	 ?s sbol:component ?comp # Searching for **sbol** components

  	 ?comp sbol:definition ?def # provides the link to that component

         filter (regex(?role, 'http://identifiers.org/so/SO:')) 

	 OPTIONAL {?def sbol:role ?role} # Defines the role

	 OPTIONAL {?def sbol:displayId ?displayId} # gives the ID of the design

	 OPTIONAL {?def dcterms:title ?title} 
  }
</code></pre><p><em>The query selects all sbol components that have an SO role type. It returns the part title, part role, part display id, part uri, and the total number of parts returned by the query. If you want to search for other types of designs, just replace SBOL with your prefered type.</em></p>
<p>To learn more about SPARQL, visit <a href="https://www.w3.org/TR/sparql11-query/">this page</a>.</p>
<h2 id="2-registering--updating-an-account">2. Registering &amp; Updating an account</h2>
<h3 id="21-registering-an-account">2.1 Registering an account</h3>
<p>Following Steps need to be followed for registering onto SynBioHub:</p>
<ol>
<li>
<p>Navigate to SynBioHub&rsquo;s home page.</p>
</li>
<li>
<p>Click on the <strong>Login or Register</strong> option on the top right hand side of your Screen.</p>
</li>
<li>
<p>Enter the following details to complete your registration process:</p>
</li>
</ol>
<ul>
<li>
<p>Full Name,</p>
</li>
<li>
<p>Affiliation(Optional),</p>
</li>
<li>
<p>Email Address,</p>
</li>
<li>
<p>Username &amp;</p>
</li>
<li>
<p>Password</p>
</li>
</ul>
<ol start="4">
<li>When you need to login into your SynBioHub account, click on the &ldquo;login or Register&rdquo; option. Then you need to enter your email address and your password to access your account.</li>
</ol>
<h3 id="22-updating-an-account">2.2 Updating an Account</h3>
<p>If you ever need to update your account, the following steps need to be followed:</p>
<ol>
<li>
<p>You need to be logged in order to update your account. Follow the previous set of steps to login, if you aren&rsquo;t logged in to your account.</p>
</li>
<li>
<p>Then, click on the <strong>profile</strong> option. It&rsquo;ll redirect you onto a page, which contains your profile.</p>
</li>
<li>
<p>On this page, update whichever section you want to, and then click on the <strong>save</strong> option at the bottom to successfully update your profile.</p>
</li>
</ol>
<h2 id="3-viewing--downloading-the-information">3. Viewing &amp; Downloading the Information</h2>
<h3 id="31-viewing-the-information">3.1 Viewing the Information</h3>
<p>There are various types of resources on SynBioHub. They&rsquo;ve been divided into the following sub-categories:</p>
<h4 id="311-viewing-a-collection">3.1.1 Viewing a collection</h4>
<ol>
<li>
<p>Navigate to SynBioHub&rsquo;s home page.</p>
</li>
<li>
<p>Then, you can search for a specific collection or select <strong>browse the collections</strong> option. It&rsquo;ll direct you to a set of collections based on what you&rsquo;ve searched for.</p>
</li>
<li>
<p>Select the collection which you want to view.</p>
</li>
<li>
<p>On the collection&rsquo;s home page, you&rsquo;ll be able to view to view the following parameters in this order:</p>
</li>
</ol>
<ul>
<li>
<p>Name: Represents the name of the collection, which you want to view.</p>
</li>
<li>
<p>ID and version: The next line represents the ID and the version of the collection. Every collection has a specific ID associated with it and a version which represents the number of times; it has been modified.</p>
</li>
<li>
<p>Resource type: This represents the type of resource. For this section, it should be a <strong>collection</strong>.</p>
</li>
<li>
<p>Created by: Displays the name of the collection&rsquo;s creator.</p>
</li>
<li>
<p>Date created: Displays the date and time when the design was created.</p>
</li>
<li>
<p>Subsequently, we have a brief description of the collection that is provided by the creator of the collection.</p>
</li>
<li>
<p>Then, we&rsquo;ve got two clickable options, that are <strong>download</strong> and <strong>back</strong>. Back navigates you to the search page and the download option consists of the ways in which you can download the collection. They&rsquo;re as follows:</p>
<table>
<thead>
<tr>
<th>Type</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>SBOL</td>
<td>The Synthetic Biology Open Language (SBOL) has been developed as a standard to support the         specification and exchange of biological design information in synthetic biology. To know more about SBOl, visit <a href="https://sbolstandard.org/datamodel-about/">this page</a></td>
</tr>
<tr>
<td>Combine Archive</td>
<td>A COMBINE archive is a single file containing the various documents (and in the future, references to documents), necessary for the description of a model and all associated data and procedures. This includes for instance, but not limited to, simulation experiment descriptions, all models needed to run the simulations and associated data files. The archive is encoded using the Open Modeling EXchange format (OMEX). To know more about Combine archive, visit <a href="http://co.mbine.org/documents/archive">this page</a>.</td>
</tr>
<tr>
<td>Annotated Genbank</td>
<td>GenBank format (GenBank Flat File Format) consists of an annotation section and a sequence section. The start of the annotation section is marked by a line beginning with the word &ldquo;LOCUS&rdquo;. The start of sequence section is marked by a line beginning with the word &ldquo;ORIGIN&rdquo; and the end of the section is marked by a line with only &ldquo;//&rdquo; To know more about Annotated genbank format, visit <a href="http://quma.cdb.riken.jp/help/gbHelp.html">this page</a>,</td>
</tr>
<tr>
<td>Annotated Plasmid Map</td>
<td>Plasmid maps are graphical representation of plasmids, that show the locations of major identifiable landmarks on DNA like restriction enzyme sites, gene of interest, plasmid name and length etc. To know more about plasmid map, visit <a href="https://en.wikipedia.org/wiki/Plasmid">this page</a></td>
</tr>
<tr>
<td>Plasmid map</td>
<td>Plasmid maps are graphical representation of plasmids, that show the locations of major identifiable landmarks on DNA like restriction enzyme sites, gene of interest, plasmid name and length etc. To know more about plasmid map, visit <a href="https://www.snapgene.com/resources/plasmid-files/">this page</a>.</td>
</tr>
<tr>
<td>Annotated zip</td>
<td>ZIP is an archive file format that supports lossless data compression. A ZIP file may contain one or more files or directories that may have been compressed. To know more about zip, visit <a href="https://en.wikipedia.org/wiki/Zip_(file_format)">this page</a></td>
</tr>
</tbody>
</table>
</li>
</ul>
<hr>
<ul>
<li>
<p>Then, you can view the members of that specific collection. There is also an option to filter the members. Click on the option to filter, and a drop down menu shall appear. You may choose the filters based on your convenience. There is also an option to search for a specific member of the collection by entering the name or ID of the member into the search box and clicking search. You may also choose the number of entries that you want to display on that page.
The list of members is sortable and can be sorted on the basis of name, identifier, type as well as description.
The list consists of the name, identifier, type as well as description of respective the members.</p>
</li>
<li>
<p>You can also view details about that specific collection, in the next section. This section contains references as well as citations.</p>
</li>
<li>
<p>Subsequently, you can also view various other properties concerning a particular collection. This specifically conatains information about the various owners of that collection.</p>
</li>
<li>
<p>Finally, you can also view the various attachments that are there in that particular collection.</p>
</li>
</ul>
<h4 id="312-viewing-an-engineered-regionpromoteractivator">3.1.2 Viewing an Engineered Region/Promoter/Activator</h4>
<ol>
<li>
<p>Navigate to SynBioHub&rsquo;s home page.</p>
</li>
<li>
<p>Then, you can search for a specific collection or select <strong>browse the collections</strong> option. It&rsquo;ll direct you to a set of collections based on what you&rsquo;ve searched for.</p>
</li>
<li>
<p>Select the collection which you want to view. Then it&rsquo;ll direct you to the collection&rsquo;s home page. You can select the resource, which you want to view.</p>
</li>
<li>
<p>On the resource&rsquo;s page, you&rsquo;ll be able to view to view the following parameters in this order:</p>
</li>
</ol>
<ul>
<li>
<p>Name: Represents the name of that particular resource.</p>
</li>
<li>
<p>ID and version: The next line represents the ID and the version of that resource. Every engineered region has a specific ID associated with it and a version which represents the number of time; it has been modified.</p>
</li>
<li>
<p>Generated from: Represents the source from where that resource had been generated from.</p>
</li>
<li>
<p>Then, we&rsquo;ve got five clickable options, that are:</p>
</li>
</ul>
<ol>
<li>
<p><strong>Download</strong></p>
</li>
<li>
<p><strong>Search</strong></p>
</li>
<li>
<p><strong>Share</strong></p>
</li>
<li>
<p><strong>Back</strong></p>
</li>
<li>
<p><strong>Add to a collection</strong></p>
</li>
</ol>
<p>The download option consists of the ways in which you can download the resource. They&rsquo;re as follows:</p>
<table>
<thead>
<tr>
<th>Type</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>SBOL</td>
<td>The Synthetic Biology Open Language (SBOL) has been developed as a standard to support the         specification and exchange of biological design information in synthetic biology. To know more about SBOl, visit <a href="https://sbolstandard.org/datamodel-about/">this page</a></td>
</tr>
<tr>
<td>Combine Archive</td>
<td>A COMBINE archive is a single file containing the various documents (and in the future, references to documents), necessary for the description of a model and all associated data and procedures. This includes for instance, but not limited to, simulation experiment descriptions, all models needed to run the simulations and associated data files. The archive is encoded using the Open Modeling EXchange format (OMEX). To know more about Combine archive, visit <a href="http://co.mbine.org/documents/archive">this page</a>.</td>
</tr>
<tr>
<td>Annotated Genbank</td>
<td>GenBank format (GenBank Flat File Format) consists of an annotation section and a sequence section. The start of the annotation section is marked by a line beginning with the word &ldquo;LOCUS&rdquo;. The start of sequence section is marked by a line beginning with the word &ldquo;ORIGIN&rdquo; and the end of the section is marked by a line with only &ldquo;//&rdquo; To know more about Annotated genbank format, visit <a href="http://quma.cdb.riken.jp/help/gbHelp.html">this page</a>,</td>
</tr>
<tr>
<td>Annotated Plasmid Map</td>
<td>Plasmid maps are graphical representation of plasmids, that show the locations of major identifiable landmarks on DNA like restriction enzyme sites, gene of interest, plasmid name and length etc. To know more about plasmid map, visit <a href="https://en.wikipedia.org/wiki/Plasmid">this page</a></td>
</tr>
<tr>
<td>Plasmid map</td>
<td>Plasmid maps are graphical representation of plasmids, that show the locations of major identifiable landmarks on DNA like restriction enzyme sites, gene of interest, plasmid name and length etc. To know more about plasmid map, visit <a href="https://www.snapgene.com/resources/plasmid-files/">this page</a>.</td>
</tr>
<tr>
<td>Annotated zip</td>
<td>ZIP is an archive file format that supports lossless data compression. A ZIP file may contain one or more files or directories that may have been compressed. To know more about zip, visit <a href="https://en.wikipedia.org/wiki/Zip_(file_format)">this page</a></td>
</tr>
<tr>
<td>GFF3</td>
<td>The GFF (General Feature Format) format consists of one line per feature, each containing 9 columns of data, plus optional track definition lines. To know more about GFF3, click <a href="http://asia.ensembl.org/info/website/upload/gff3.html">here</a>.</td>
</tr>
<tr>
<td>FASTA</td>
<td>In bioinformatics and biochemistry, the FASTA format is a text-based format for representing either nucleotide sequences or amino acid (protein) sequences, in which nucleotides or amino acids are represented using single-letter codes. The format also allows for sequence names and comments to precede the sequences. The format originates from the FASTA software package, but has now become a near universal standard in the field of bioinformatics. To know more about FATSA file format, click <a href="https://en.wikipedia.org/wiki/FASTA_format">here</a></td>
</tr>
<tr>
<td>Image</td>
<td>An image is an artifact that depicts visual perception, such as a photograph or other two-dimensional picture, that resembles a subject—usually a physical object—and thus provides a depiction of it. In the context of signal processing, an image is a distributed amplitude of color(s). A pictorial script is a writing system that employs images as symbols for various semantic entities, rather than the abstract signs used by alphabets</td>
</tr>
</tbody>
</table>
<hr>
<p>Then is the <strong>search</strong> option. There are three ways you can use the search option. They&rsquo;re as follows:</p>
<ol>
<li>
<p>Find Uses: This returns any other object that refers to this object that the user has given as an input, for example, if this is an engineered region, then it will return all other components that use this as a sub-component.</p>
</li>
<li>
<p>Find Twins:  This returns other components, if and only if they have the same sequence as the input by the user.</p>
</li>
<li>
<p>Find Similar: This returns other components that have similar sequences.</p>
</li>
</ol>
<p>The third option is that of <strong>sharing</strong> your resource. There are 2 ways by which you can share your resource. They&rsquo;re as follows:</p>
<ol>
<li>
<p>Send to Benchling: For using Benchling to share your resource, you need to configure a Benchling remote. The Benchling Life Sciences R&amp;D Cloud is an informatics platform to accelerate, measure, and forecast R&amp;D from discovery through bioprocessing. To know more about Benchling and how you can configure a benching remote, click <a href="https://www.benchling.com/">here</a>.</p>
</li>
<li>
<p>Send to ICE: ICE is a registry platform that provides robust data storage for DNA components, integrated tools for part characterization, as well as mechanisms for secure access and information sharing with other users and software tools.
ICE is built upon the idea of a Web of Registries, and thus provides strong support for distributed, interconnected use. It includes several useful graphical applications for sequence annotation, manipulation and analysis. To know more about ICE, click <a href="https://public-registry.jbei.org/login">here</a>.</p>
</li>
</ol>
<p>Fourth is the <strong>back</strong> option. This option navigates back to the main page of the collection, which that particular resource is a part of.</p>
<p>Finally, we have the <strong>Add to collection</strong> option. This option adds your resource to a specific collection. Clicking on this option, would take you to a page that shall display the collections, that you would&rsquo;ve created. Select the collection, to which you want to add this engineered region to and then click on add to collection. This shall add your resource to the selected collection.</p>
<ul>
<li>
<p>Next we have is the VisBOL Navigator, with zooming capabilities. This, graphically represents the structure of the engineered region. By hovering your pointer over this structure, you can view the following attributes of that particular engineered region: type of resource, identifier, name and role of the engineered region.</p>
</li>
<li>
<p>Then we have the details of that engineered region. This consists of the following:</p>
</li>
</ul>
<ol>
<li>
<p>Type: This specifies the category of biochemical or physical entity. For example DNA, protein, or small molecule that a ComponentDefinition object abstracts for the purpose of engineering design. To know more about Type, click <a href="https://dissys.github.io/sbol-owl/sbol-owl.html#type">here</a>.</p>
</li>
<li>
<p>Role: Clarifies the potential function of an entity in a biochemical or physical context. When it is used for ComponentDefinitions, it MUST identify terms from ontologies that are consistent with the types property of the ComponentDefinition.To know more about role, click <a href="https://dissys.github.io/sbol-owl/sbol-owl.html#role">here</a>.</p>
</li>
<li>
<p>Sequence: The purpose of the Sequence class is to represent the primary structure of a ComponentDefinition object and the manner in which it is encoded. This representation is accomplished by means of the elements property and encoding property. To know more about Sequences, click <a href="https://dissys.github.io/sbol-owl/sbol-owl.html#Sequence">here</a>.</p>
</li>
<li>
<p>Also, you can view reference as well as citations.</p>
</li>
</ol>
<ul>
<li>Then, we have the section named, <strong>other properties</strong>. This consists of the following attributes:</li>
</ul>
<ol>
<li>
<p>Ontology for Biomedical Investigations(OBI): A centrally registered identifier symbol used to uniquely identify objects given by International DOI Foundation. The DOI system is particularly used for electronic documents such as journal articles. To know more about OBI, click <a href="http://www.ontobee.org/ontology/OBI?iri=http://purl.obolibrary.org/obo/OBI_0002110">here</a>.</p>
</li>
<li>
<p>data_1179</p>
</li>
<li>
<p>SynBioHub#Owned_by: Username of the design owner.</p>
</li>
<li>
<p>SynBioHub#TopLevel</p>
</li>
</ol>
<ul>
<li>
<p>Subsequently, we have the section titled as, <strong>Member of these collections</strong>. This section contains the list of collection&rsquo;s to which that particular resource belongs to.</p>
</li>
<li>
<p>Then, we have the <strong>attachments</strong> page. In this section, you can view the various attachments, of a resource.</p>
</li>
<li>
<p>The next 3 sections are <strong>iGEM main page</strong>, <strong>iGEM design page</strong> and the <strong>iGEM experience page</strong>. Then is the section that contains, <strong>Sequence Visualisation</strong> of that particular resource. Finally, we have the <strong>component sankey</strong> and the <strong>component bar</strong>. The iGEM parts are plugins, hence they are not part of every instance and the Sequence Visualization, Sankey, and Component bar are additional plugins, hence they&rsquo;ve been explained in detail in the <a href="https://synbiohub.github.io/plugins/">plugins section</a>.</p>
</li>
</ul>
<h3 id="32-downloading-the-information">3.2 Downloading the Information</h3>
<h4 id="321-downloading-a-collection">3.2.1 Downloading a collection</h4>
<ol>
<li>
<p>Navigate to SynBioHub&rsquo;s home page.</p>
</li>
<li>
<p>Then, you can search for a specific collection or select <strong>browse the collections</strong> option. It&rsquo;ll direct you to a set of collections based on what you&rsquo;ve searched for.</p>
</li>
<li>
<p>Select the collection which you want to download. Then it&rsquo;ll direct you to the collection&rsquo;s home page.</p>
</li>
<li>
<p>On the left there would be an option to download the collection, in drop down format. Following are the types in which you can download the collection:</p>
</li>
</ol>
<table>
<thead>
<tr>
<th>Type</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>SBOL</td>
<td>The Synthetic Biology Open Language (SBOL) has been developed as a standard to support the         specification and exchange of biological design information in synthetic biology. To know more about SBOl, visit <a href="https://sbolstandard.org/datamodel-about/">this page</a></td>
</tr>
<tr>
<td>Combine Archive</td>
<td>A COMBINE archive is a single file containing the various documents (and in the future, references to documents), necessary for the description of a model and all associated data and procedures. This includes for instance, but not limited to, simulation experiment descriptions, all models needed to run the simulations and associated data files. The archive is encoded using the Open Modeling EXchange format (OMEX). To know more about Combine archive, visit <a href="http://co.mbine.org/documents/archive">this page</a>.</td>
</tr>
<tr>
<td>Annotated Genbank</td>
<td>GenBank format (GenBank Flat File Format) consists of an annotation section and a sequence section. The start of the annotation section is marked by a line beginning with the word &ldquo;LOCUS&rdquo;. The start of sequence section is marked by a line beginning with the word &ldquo;ORIGIN&rdquo; and the end of the section is marked by a line with only &ldquo;//&rdquo; To know more about Annotated genbank format, visit <a href="http://quma.cdb.riken.jp/help/gbHelp.html">this page</a>,</td>
</tr>
<tr>
<td>Annotated Plasmid Map</td>
<td>Plasmid maps are graphical representation of plasmids, that show the locations of major identifiable landmarks on DNA like restriction enzyme sites, gene of interest, plasmid name and length etc. To know more about plasmid map, visit <a href="https://en.wikipedia.org/wiki/Plasmid">this page</a></td>
</tr>
<tr>
<td>Plasmid map</td>
<td>Plasmid maps are graphical representation of plasmids, that show the locations of major identifiable landmarks on DNA like restriction enzyme sites, gene of interest, plasmid name and length etc. To know more about plasmid map, visit <a href="https://www.snapgene.com/resources/plasmid-files/">this page</a>.</td>
</tr>
<tr>
<td>Annotated zip</td>
<td>ZIP is an archive file format that supports lossless data compression. A ZIP file may contain one or more files or directories that may have been compressed. To know more about zip, visit <a href="https://en.wikipedia.org/wiki/Zip_(file_format)">this page</a></td>
</tr>
</tbody>
</table>
<hr>
<h4 id="322-downloading-a-module-engineered-region-promoter-activator">3.2.2 Downloading a Module/ Engineered Region/ Promoter/ Activator</h4>
<ol>
<li>
<p>Navigate to SynBioHub&rsquo;s home page.</p>
</li>
<li>
<p>Then, you can search for a specific collection or select <strong>browse the collections</strong> option. It&rsquo;ll direct you to a set of collections based on what you&rsquo;ve searched for.</p>
</li>
<li>
<p>Select the collection which you want to download. Then it&rsquo;ll direct you to the collection&rsquo;s home page. From the collection&rsquo;s home page select resource which you want to download and click on that. This will direct you to the resource&rsquo;s home page.</p>
</li>
<li>
<p>Then, on the left there would be an option to download that resource, in drop down format. Following are the types in which you can download the collection:</p>
</li>
</ol>
<table>
<thead>
<tr>
<th>Type</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>SBOL</td>
<td>The Synthetic Biology Open Language (SBOL) has been developed as a standard to support the         specification and exchange of biological design information in synthetic biology. To know more about SBOl, visit <a href="https://sbolstandard.org/datamodel-about/">this page</a></td>
</tr>
<tr>
<td>Combine Archive</td>
<td>A COMBINE archive is a single file containing the various documents (and in the future, references to documents), necessary for the description of a model and all associated data and procedures. This includes for instance, but not limited to, simulation experiment descriptions, all models needed to run the simulations and associated data files. The archive is encoded using the Open Modeling EXchange format (OMEX). To know more about Combine archive, visit <a href="http://co.mbine.org/documents/archive">this page</a>.</td>
</tr>
<tr>
<td>Annotated Genbank</td>
<td>GenBank format (GenBank Flat File Format) consists of an annotation section and a sequence section. The start of the annotation section is marked by a line beginning with the word &ldquo;LOCUS&rdquo;. The start of sequence section is marked by a line beginning with the word &ldquo;ORIGIN&rdquo; and the end of the section is marked by a line with only &ldquo;//&rdquo; To know more about Annotated genbank format, visit <a href="http://quma.cdb.riken.jp/help/gbHelp.html">this page</a>,</td>
</tr>
<tr>
<td>Annotated Plasmid Map</td>
<td>Plasmid maps are graphical representation of plasmids, that show the locations of major identifiable landmarks on DNA like restriction enzyme sites, gene of interest, plasmid name and length etc. To know more about plasmid map, visit <a href="https://en.wikipedia.org/wiki/Plasmid">this page</a></td>
</tr>
<tr>
<td>Plasmid map</td>
<td>Plasmid maps are graphical representation of plasmids, that show the locations of major identifiable landmarks on DNA like restriction enzyme sites, gene of interest, plasmid name and length etc. To know more about plasmid map, visit <a href="https://www.snapgene.com/resources/plasmid-files/">this page</a>.</td>
</tr>
<tr>
<td>Annotated zip</td>
<td>ZIP is an archive file format that supports lossless data compression. A ZIP file may contain one or more files or directories that may have been compressed. To know more about zip, visit <a href="https://en.wikipedia.org/wiki/Zip_(file_format)">this page</a></td>
</tr>
<tr>
<td>GFF3</td>
<td>The GFF (General Feature Format) format consists of one line per feature, each containing 9 columns of data, plus optional track definition lines. To know more about GFF3, click <a href="http://asia.ensembl.org/info/website/upload/gff3.html">here</a>.</td>
</tr>
<tr>
<td>FASTA</td>
<td>In bioinformatics and biochemistry, the FASTA format is a text-based format for representing either nucleotide sequences or amino acid (protein) sequences, in which nucleotides or amino acids are represented using single-letter codes. The format also allows for sequence names and comments to precede the sequences. The format originates from the FASTA software package, but has now become a near universal standard in the field of bioinformatics. To know more about FATSA file format, click <a href="https://en.wikipedia.org/wiki/FASTA_format">here</a></td>
</tr>
<tr>
<td>Image</td>
<td>An image is an artifact that depicts visual perception, such as a photograph or other two-dimensional picture, that resembles a subject—usually a physical object—and thus provides a depiction of it. In the context of signal processing, an image is a distributed amplitude of color(s). A pictorial script is a writing system that employs images as symbols for various semantic entities, rather than the abstract signs used by alphabets</td>
</tr>
<tr>
<td>Annotated Zip</td>
<td>ZIP is an archive file format that supports lossless data compression. A ZIP file may contain one or more files or directories that may have been compressed. To know more about zip, visit <a href="https://en.wikipedia.org/wiki/Zip_(file_format)">this page</a></td>
</tr>
</tbody>
</table>
<hr>
<h2 id="4-submitting-managing--updating-submissions">4. Submitting, Managing &amp; Updating Submissions</h2>
<h3 id="41-submitting">4.1 Submitting</h3>
<p>For submitting your designs to SynBioHub, following steps need to be followed:</p>
<ol>
<li>
<p>Navigate to SynBioHub&rsquo;s home page.</p>
</li>
<li>
<p>Click on the <strong>Submit</strong> option and it&rsquo;ll direct you to the Submit page.</p>
</li>
</ol>
<p>Now, there are 2 ways you make your submissions i.e, creating a new collection and then submitting your resource or submitting to an existing collection.</p>
<h4 id="411-creating-a-new-collection">4.1.1 Creating a new collection</h4>
<ol>
<li>
<p>Select the <strong>create a new collection</strong> option.</p>
</li>
<li>
<p>Following are the attributes that need to be filled up :</p>
</li>
</ol>
<ul>
<li>
<p><strong>Name</strong>: This is the name string that&rsquo;ll be assigned to the submission.</p>
</li>
<li>
<p><strong>Collection Description</strong>: This is the description string that briefly explains your submission.</p>
</li>
<li>
<p><strong>Collection ID</strong>: This is a user-defined string identifier for the submission and it should contain alphanumeric and underscores characters only.</p>
</li>
<li>
<p><strong>Collection Version</strong>: This is the version string which is associated with the submission.</p>
</li>
<li>
<p><strong>Citations</strong> (optional): This is a list of comma separated pubmed IDs of citations to store with the submission, just for reference. It is an optional attribute.</p>
</li>
<li>
<p><strong>SBOL/GENBANK/GFF3/FASTA/ZIP file</strong> (OPTIONAL): You can optionally submit a file. The file should be of the following types:</p>
</li>
</ul>
<table>
<thead>
<tr>
<th>Type</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>SBOL</td>
<td>The Synthetic Biology Open Language (SBOL) has been developed as a standard to support the         specification and exchange of biological design information in synthetic biology. To know more about SBOl, visit <a href="https://sbolstandard.org/datamodel-about/">this page</a></td>
</tr>
<tr>
<td>Annotated zip</td>
<td>ZIP is an archive file format that supports lossless data compression. A ZIP file may contain one or more files or directories that may have been compressed. To know more about zip, visit <a href="https://en.wikipedia.org/wiki/Zip_(file_format)">this page</a></td>
</tr>
<tr>
<td>GFF3</td>
<td>The GFF (General Feature Format) format consists of one line per feature, each containing 9 columns of data, plus optional track definition lines. To know more about GFF3, click <a href="http://asia.ensembl.org/info/website/upload/gff3.html">here</a>.</td>
</tr>
<tr>
<td>FASTA</td>
<td>In bioinformatics and biochemistry, the FASTA format is a text-based format for representing either nucleotide sequences or amino acid (protein) sequences, in which nucleotides or amino acids are represented using single-letter codes. The format also allows for sequence names and comments to precede the sequences. The format originates from the FASTA software package, but has now become a near universal standard in the field of bioinformatics. To know more about FATSA file format, click <a href="https://en.wikipedia.org/wiki/FASTA_format">here</a></td>
</tr>
<tr>
<td>Zip</td>
<td>ZIP is an archive file format that supports lossless data compression. A ZIP file may contain one or more files or directories that may have been compressed. To know more about zip, visit <a href="https://en.wikipedia.org/wiki/Zip_(file_format)">this page</a></td>
</tr>
</tbody>
</table>
<hr>
<ul>
<li>
<p><strong>Handler</strong>: You also need to select a handler. Following options are available:</p>
<ul>
<li>
<p>Default</p>
</li>
<li>
<p>Excel Library</p>
</li>
<li>
<p>Excel Composition</p>
</li>
<li>
<p>Snapgene</p>
</li>
</ul>
</li>
<li>
<p>Finally, you have an option to overwrite the existing objects in that collection i.e, removing all the existing objects in the collection and adding new ones or keeping the old resources and adding the new ones.</p>
</li>
</ul>
<ol start="3">
<li>At last, click on <strong>submit</strong>.</li>
</ol>
<h4 id="412-submitting-to-an-existing-collection">4.1.2 Submitting to an existing collection</h4>
<ol>
<li>
<p>Select the <strong>add to existing collection</strong> option.</p>
</li>
<li>
<p>Then, select your desired collection, to which you want to add your submission from the dropdown menu.</p>
</li>
<li>
<p>Subsequently, upload the files which you want to submit. The file should be of the following types:</p>
</li>
</ol>
<table>
<thead>
<tr>
<th>Type</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>SBOL</td>
<td>The Synthetic Biology Open Language (SBOL) has been developed as a standard to support the         specification and exchange of biological design information in synthetic biology. To know more about SBOl, visit <a href="https://sbolstandard.org/datamodel-about/">this page</a></td>
</tr>
<tr>
<td>Annotated zip</td>
<td>ZIP is an archive file format that supports lossless data compression. A ZIP file may contain one or more files or directories that may have been compressed. To know more about zip, visit <a href="https://en.wikipedia.org/wiki/Zip_(file_format)">this page</a></td>
</tr>
<tr>
<td>GFF3</td>
<td>The GFF (General Feature Format) format consists of one line per feature, each containing 9 columns of data, plus optional track definition lines. To know more about GFF3, click <a href="http://asia.ensembl.org/info/website/upload/gff3.html">here</a>.</td>
</tr>
<tr>
<td>FASTA</td>
<td>In bioinformatics and biochemistry, the FASTA format is a text-based format for representing either nucleotide sequences or amino acid (protein) sequences, in which nucleotides or amino acids are represented using single-letter codes. The format also allows for sequence names and comments to precede the sequences. The format originates from the FASTA software package, but has now become a near universal standard in the field of bioinformatics. To know more about FATSA file format, click <a href="https://en.wikipedia.org/wiki/FASTA_format">here</a></td>
</tr>
<tr>
<td>Zip</td>
<td>ZIP is an archive file format that supports lossless data compression. A ZIP file may contain one or more files or directories that may have been compressed. To know more about zip, visit <a href="https://en.wikipedia.org/wiki/Zip_(file_format)">this page</a></td>
</tr>
</tbody>
</table>
<hr>
<ol start="4">
<li>
<p>Finally, you have an option to overwrite the existing objects in that collection i.e, removing all the existing objects in the collection and adding new ones or keeping the old resources and adding the new ones.</p>
</li>
<li>
<p>At last, click on <strong>submit</strong>.</p>
</li>
</ol>
<h3 id="42-managing">4.2 Managing</h3>
<ol>
<li>
<p>Navigate to SynBioHub&rsquo;s home page.</p>
</li>
<li>
<p>Click on <strong>submissions</strong>. It&rsquo;ll direct you to a page titled <strong>manage submissions</strong>. On this page you&rsquo;ll be able to see collections to be divided, broadly under 2 categories. The 1st one being, <strong>private collections</strong>, which you&rsquo;ve created and only you have access to them and 2nd one being, <strong>public collections</strong>, to which all the users have access to.</p>
</li>
<li>
<p>You can view your collection by clicking on the collection name.</p>
</li>
<li>
<p>In order to make your collection public, click on <strong>make public</strong>. This shall enable that other users to view your collection and its contents.</p>
</li>
<li>
<p>To delete your collection, click on <strong>Remove</strong>. This will remove the collection as well as the resources contained, in that specific collection, namely promoters, activators, modules, components, engineered region, etc.</p>
</li>
</ol>
<h3 id="43-updating">4.3 Updating</h3>
<ol>
<li>
<p>On the submit page, click on <em>Submit to an existing collection</em> option.</p>
</li>
<li>
<p>Then, select to the collection you want to submit your design to.</p>
</li>
<li>
<p>Select the design that you want to submit.</p>
</li>
<li>
<p>You may choose to <em>overwrite the existing objects</em> in a collection.</p>
</li>
<li>
<p>Then, click on submit.</p>
</li>
</ol>
<h2 id="5-editing-a-submission">5. Editing a Submission</h2>
<p>To edit your submissions, following steps need to be followed:</p>
<ol>
<li>
<p>Go to the ubmissions page.</p>
</li>
<li>
<p>Select the collection that you want to edit.</p>
</li>
<li>
<p>You can edit the name &amp; description of the resource as well as add a source.</p>
</li>
<li>
<p>If you want to add a detailed description, source, reference or notes for that very resource, just Scroll down to find those options under the <strong>details</strong> section.</p>
</li>
<li>
<p>Subsequently, scroll down to find the members of the collection. You may click on the <em>delete</em> sign just next to the description of the design, if you want to delete it or click on the member&rsquo;s name if you want to view its details.</p>
</li>
<li>
<p>Steps 3 &amp; 4 can be followed, if you want to edit any of the any details of the member&rsquo;s of that particular.</p>
</li>
</ol>
<h2 id="6-data-sharing-mechanisms">6. Data Sharing Mechanisms</h2>
<p>There are a couple of ways to share data in SynBioHub. They&rsquo;re mentioned as follows:</p>
<h3 id="61-using-the-share-option">6.1 Using the Share option</h3>
<ol>
<li>
<p>Select the collection/design that you want to share.</p>
</li>
<li>
<p>Click the <strong>share</strong> option. Once you click on the share option, a link shall appear.</p>
</li>
<li>
<p>Now, copy that link &amp; you can share the link, thereby sharing the resource.</p>
</li>
</ol>
<h3 id="62-adding-an-owner">6.2 Adding an owner</h3>
<p>You can also share your design by giving any number of persons access by adding them as an owner for your design. This will give them full administrative access to your resources.
The steps to do so are as follows:</p>
<ol>
<li>
<p>Select the collection/design that you want to share.</p>
</li>
<li>
<p>Click on the <strong>add an owner</strong> option.</p>
</li>
<li>
<p>Select the Username of the person, with whom you want to share your resources.</p>
</li>
<li>
<p>Click on <strong>grant ownership</strong>.</p>
</li>
</ol>
<h2 id="7-administration">7. Administration</h2>
<p>Note: <strong>The administration section will only be available, if you&rsquo;re using a SynBioHub local instance. If you&rsquo;re using a global instance, such as <a href="https://synbiohub.org/">this</a> one, then the administration section won&rsquo;t be available unless you have privileged access to SynBioHub.</strong></p>
<p>To access the admin page, the following steps need to be followed:</p>
<ol>
<li>
<p>Navigate to the home page of your SynbioHub local instance.</p>
</li>
<li>
<p>Click on <strong>admin</strong>, in order to access the administrative settings of your instance. This will direct you to the administrative page. It has been divided into several sub-sections, for your convenience. They&rsquo;re discussed in detail, as follows:</p>
</li>
</ol>
<h4 id="71-status">7.1 Status</h4>
<p>Several parameters essential for the smooth functioning of SynBioHub, are mentioned in this sub-section. They&rsquo;re as follows:</p>
<table>
<thead>
<tr>
<th>parameter</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>1. <strong>Platform</strong>: Linux x64; 4.19.76-linuxkit</td>
<td>This parameter represents the platform on which SynBioHub was built.</td>
</tr>
<tr>
<td>2. <strong>Node Version</strong>: vXX.X.X</td>
<td>This represents the node.js version, on which your local instance is running</td>
</tr>
<tr>
<td>3. <strong>Instance Name</strong>:  name_of_your_choice</td>
<td>This represents the name of your SynBioHub instance. This can be changed based on your convenience.</td>
</tr>
<tr>
<td>4. <strong>Listen Port: 7777</strong></td>
<td>When Oracle HTTP Server is started, by default, it listens for requests on port 7777 (non-SSL). If port 7777 is occupied, Oracle HTTP Server listens on the next available port number between a range of 7777-7877. Thus, if port 7777 is busy, it would listen on port 7778, and so on. To get more information on this, click <a href="https://docs.oracle.com/cd/B12037_01/server.101/b12255/netconf.htm#:~:text=When%20Oracle%20HTTP%20Server%20is,port%207778%2C%20and%20so%20on.">here</a>.</td>
</tr>
<tr>
<td>5. <strong>SPARQL Endpoint</strong>: http://virtuoso:8890/sparql</td>
<td>A SPARQL Endpoint is a Point of Presence on an HTTP network that&rsquo;s capable of receiving and processing SPARQL Protocol requests. It is identified by a URL commonly referred to as a SPARQL Endpoint URL. To know more about SPARQl endpoints, click <a href="https://www.w3.org/wiki/SparqlEndpointDescription">here</a>.</td>
</tr>
<tr>
<td>6. <strong>Graph Store Endpoint</strong>: http://virtuoso:8890/sparql-graph-crud-auth/</td>
<td></td>
</tr>
<tr>
<td>7. <strong>Default Graph</strong>: http://localhost:7777/public</td>
<td></td>
</tr>
<tr>
<td>8. <strong>Graph Prefix</strong>: http://localhost:7777/</td>
<td></td>
</tr>
<tr>
<td>9. <strong>Thread Pool Size</strong>:</td>
<td></td>
</tr>
<tr>
<td>10. <strong>Upload Limit</strong>: X mb</td>
<td>Shows the maximum limit of files that you can upload at once to your instance</td>
</tr>
</tbody>
</table>
<h4 id="72-graphs">7.2 Graphs</h4>
<p>This section basically displays the graph URI and its corresponding triples. You can select the number of graphs you want to view on one page as well as search for any specific graphs, with the help of search box. You can also sort the graphs on the basis of number of triples.</p>
<h4 id="73-log">7.3 Log</h4>
<p>This section displays the log file for your SynBioHub local instance. You can view 4 types of logs, which are as follows:</p>
<ol>
<li>
<p><strong>Error</strong></p>
</li>
<li>
<p><strong>Warn</strong></p>
</li>
<li>
<p><strong>Info</strong></p>
</li>
<li>
<p><strong>Debug</strong></p>
</li>
</ol>
<p>You may use the pointer to view differs types of logs.</p>
<h4 id="74-mail">7.4 Mail</h4>
<p>On this page, you have 2 options, that are sendgrid API key and sendgrid from email.</p>
<p>In the first column, you need to fill the API key, that you&rsquo;d need to generate from sendgrid. SendGrid is a cloud-based SMTP provider that allows you to send email without having to maintain email servers. SendGrid manages all of the technical details, from scaling the infrastructure to ISP outreach and reputation monitoring to whitelist services and real time analytics. To know more about sendgrid, click <a href="https://sendgrid.com/wp-content/uploads/2016/09/SendGrid-Implementation-Review.pdf">here</a>.</p>
<ol>
<li><strong>SendGrid API key</strong>: For generating an API key, you need to have a sendgrid account. To sign-up for sendgrid and hence generating an API key, click <a href="https://synbiohub.github.io/setup/#setting-up-sendgrid">here</a>.</li>
<li><strong>SendGrid From Email</strong>:  Here, you can enter the email, which you can use for sending emails.</li>
</ol>
<h4 id="75-plugins">7.5 Plugins</h4>
<p>On this page, you can configure the plugins which you want to use. Plugins are modular stand-alone additions to SynBioHub. They function in a way that is similar to browser extensions. They can be installed separately from the browser/SynBioHub and provide additional ‘custom’ functionality to the browser/SynBioHub experience despite having a completely separate code base from the browser/SynBioHub. Though, they seem integrated to the user.</p>
<p>They&rsquo;re broadly classified into 3 types, which are as follows:</p>
<ol>
<li>
<p>Submit: Submit plugins are available to use from the submit endpoint. They work by taking in the file that is uploaded in the submit and processing it to return SBOL to the SynBioHub endpoint.</p>
</li>
<li>
<p>Visual: Visual plugins are available on all ‘endpoint’ pages, for example pages for components, sequences, activities, etc. Visual plugins return html to be displayed on the page.</p>
</li>
<li>
<p>Download: Download plugins are available on all ‘endpoint’ pages, for example pages for components, sequences, activities, etc. Download plugins return some kind of file which is downloaded by the user.</p>
</li>
</ol>
<p>To know more about plugins in detail, visit the <a href="https://synbiohub.github.io/plugins/">plugins section</a>.</p>
<p>As mentioned above, the plugins are divided into 3 categories all having the same attributes. The attributes, which need to be filled up are as follows:</p>
<ul>
<li>
<p>Name: This is the name which you want to give to your plugin.</p>
</li>
<li>
<p>URL: In this column, you need to enter the URL for your plugin.</p>
</li>
</ul>
<p>After filling both of the columns click on <strong>save</strong>, in order to save your plugin.</p>
<h4 id="76-registries">7.6 Registries</h4>
<p>SynBioHub also supports the Web of Registries concept i.e, the idea of multiple separate repositories linked together by shared common semantics. In the case of SynBioHub, SBOL is used to support the common exchange of data, thus positioning SynBioHub to support the development of synthetic biology workflows by acting as a source and a storage facility for designs.</p>
<p>Hence, Web Of registries enables communication between SynBioHub instances. Sharing designs between instances of SynBioHub is also further facilitated via the Web of Registries service. Federated SynBioHub instances can reference parts in the public graph of other federated instances. In order to support this federation, the Web of Registries service maintains information about all SynBioHub registries, such as their name, description, administrator email, URI prefix, and uniform resource locator(URL).</p>
<p>To know more about web-of-registries, visit <a href="https://synbiohub.github.io/setup/#web-of-registries-in-synbiohub">this</a> page.</p>
<ul>
<li>
<p>On this page, you can configure the registry for your local instance. To configure the registry for your local instance, enter the administrator email in the column just besides the web-of-registries link. After entering the admin email, click on <strong>submit</strong> and this would configure the admin email for the registries.</p>
</li>
<li>
<p>There are 2 options given just below the local web of registries sub-section, which are:</p>
</li>
</ul>
<ol>
<li>
<p><strong>Retrieve from web of registries</strong>: This option shall retrieve the web of registries that are already existing globally, locally onto you instance, if in case you delete them.</p>
</li>
<li>
<p><strong>Add registry</strong>: This shall enable you to add a registry onto your instance. For adding, click on this option. Then, a popup shall appear. Subsequently, enter the registry URI prefix and the Registry URL. Finally, click on <strong>submit</strong>.</p>
</li>
</ol>
<ul>
<li>Several local web-of-registries are also given on this page alongside their Synbiohub URL. Here, you have an option to <strong>save</strong> as well as <strong>delete</strong> the already existing registries.</li>
</ul>
<h4 id="77-remotes">7.7 Remotes</h4>
<p>There are 2 types of remotes you can configure for SynBioHub i.e, <strong>Benchling</strong> and <strong>ICE</strong>. They&rsquo;re discussed in detail as follows:</p>
<ol>
<li>
<p><strong>Benchling</strong>: Benchling is an integrated software solution for experiment design, note-taking, and molecular biology. It is a free, intelligent research platform with tools for note-taking, molecular biology, and sample tracking. Benchling includes an electronic lab notebook, a molecular biology suite for design and analysis, and a bioregistry to track inventory. To know more about benching, click <a href="http://benchling.com/academic">here</a>.</p>
</li>
<li>
<p><strong>ICE</strong>: ICE is a registry platform that provides robust data storage for DNA components, integrated tools for part characterization, as well as mechanisms for secure access and information sharing with other users and software tools. To know more about ICE, click <a href="https://ice.jbei.org/">here</a>.</p>
</li>
</ol>
<p>Upon opening this page for the first time you&rsquo;ll see a message stating: <em>No remotes configured</em>. To configure a remote for your instance, follow the following steps:</p>
<h5 id="771-configuring-a-benchling-remote">7.7.1 Configuring a benchling remote</h5>
<ol>
<li>
<p>When you&rsquo;ll click on the type box, a dropdown menu shall appear containing 2 options, ice and benchling. For configuring a benchling remote, select <strong>benchling</strong>. Then click on <strong>add remote</strong>.</p>
</li>
<li>
<p>As soon as you&rsquo;ll click on add remote, a popup shall appear asking you to fill up certain set of attributes. They&rsquo;re as follows:</p>
</li>
</ol>
<table>
<thead>
<tr>
<th>Attribute</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>1. ID</td>
<td>Enter ID of the benchling remote.</td>
</tr>
<tr>
<td>2. Type</td>
<td>This shall display the type of remote that you would&rsquo;ve chosen, initially.</td>
</tr>
<tr>
<td>3. URL</td>
<td>Enter the URL for the benchling remote.</td>
</tr>
<tr>
<td>4. Sequence Suffix</td>
<td>This represents the suffix to use for the sequences found on benchling.</td>
</tr>
<tr>
<td>5. Default Folder ID</td>
<td>Enter the default folder on benchling to access.</td>
</tr>
<tr>
<td>6. Public</td>
<td>Should be checked, if you want the remote to be visible publicly. If not, then it should not be checked.</td>
</tr>
<tr>
<td>7. Reject Unauthorized</td>
<td>Should be checked, if you want to the SSL certificates to be checked. If not, then it should not be checked.</td>
</tr>
<tr>
<td>8. Folder Prefix</td>
<td>Represents the prefix to use for folders on benchling.</td>
</tr>
<tr>
<td>9. Root Collection Display ID</td>
<td>Represents the display id for the root collection on the remote.</td>
</tr>
<tr>
<td>10. Root Collection Name</td>
<td>This represents the name for the root collection on the remote.</td>
</tr>
<tr>
<td>11. Root Collection Description</td>
<td>A brief description about the root collection on the remote.</td>
</tr>
<tr>
<td>12. Benchling API Token</td>
<td>The token generated by benchling should be entered here.</td>
</tr>
</tbody>
</table>
<hr>
<p>Once you are through with all the attributes, click on submit.</p>
<ol start="3">
<li>Once you&rsquo;ve clicked on submit, you&rsquo;ll be able to see a fully configured remote on the left. You can also delete and edit your remote. To delete any of your configured remotes, click on delete. To edit any of the information, click on edit. Once you&rsquo;ve edited the information, click on submit.</li>
</ol>
<h5 id="712-configuring-an-ice-remote">7.1.2 Configuring an ICE remote</h5>
<ol>
<li>
<p>When you&rsquo;ll click on the type box, a dropdown menu shall appear containing 2 options, ice and benchling. For configuring an ICE remote, select <strong>ICE</strong> and then click on <strong>add remote</strong>.</p>
</li>
<li>
<p>As soon as you&rsquo;ll click on add remote, a popup shall appear asking you to fill up certain set of attributes. They&rsquo;re as follows:</p>
</li>
</ol>
<table>
<thead>
<tr>
<th>Attribute</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>1. ID</td>
<td>Enter ID of the benchling remote.</td>
</tr>
<tr>
<td>2. Type</td>
<td>This shall display the type of remote that you would&rsquo;ve chosen, initially.</td>
</tr>
<tr>
<td>3. URL</td>
<td>Enter the URL for the benchling remote.</td>
</tr>
<tr>
<td>4. Sequence Suffix</td>
<td>This represents the suffix to use for the sequences found on benchling.</td>
</tr>
<tr>
<td>5. Default Folder ID</td>
<td>Enter the default folder on benchling to access.</td>
</tr>
<tr>
<td>6. Public</td>
<td>Should be checked, if you want the remote to be visible publicly. If not, then it should not be checked.</td>
</tr>
<tr>
<td>7. Reject Unauthorized</td>
<td>Should be checked, if you want to the SSL certificates to be checked. If not, then it should not be checked.</td>
</tr>
<tr>
<td>8. Folder Prefix</td>
<td>Represents the prefix to use for folders on benchling.</td>
</tr>
<tr>
<td>9. Root Collection Display ID</td>
<td>Represents the display id for the root collection on the remote.</td>
</tr>
<tr>
<td>10. Root Collection Name</td>
<td>This represents the name for the root collection on the remote.</td>
</tr>
<tr>
<td>11. Root Collection Description</td>
<td>A brief description about the root collection on the remote.</td>
</tr>
<tr>
<td>12. ICE API Token</td>
<td>The token generated by ICE should be entered here.</td>
</tr>
<tr>
<td>13. ICE API Token Client</td>
<td></td>
</tr>
<tr>
<td>14. ICE API Token Owner</td>
<td></td>
</tr>
<tr>
<td>15. ICE Collection</td>
<td>Name of the ICE collection.</td>
</tr>
<tr>
<td>16. Group ID</td>
<td></td>
</tr>
<tr>
<td>17. PI</td>
<td>Principal Investigator name should be entered.</td>
</tr>
<tr>
<td>18. PI Email</td>
<td>Principal Investigator email should be entered.</td>
</tr>
<tr>
<td>19. Part Number Prefix</td>
<td>Prefix that is to be used for the parts.</td>
</tr>
</tbody>
</table>
<hr>
<ol start="3">
<li>Once you&rsquo;ve clicked on submit, you&rsquo;ll be able to see a fully configured remote on the left. You can also delete and edit your remote. To delete any of your configured remotes, click on delete. To edit any of the information, click on edit. Once you&rsquo;ve edited the information, click on submit.</li>
</ol>
<h4 id="78-sbolexplorer">7.8 SBOLExplorer</h4>
<p>SBOLExplorer is a service that simplifies the process of analysing and searching for parts within genetic design repositories.</p>
<p>Here, you can enter and save your SBOLExplorer endpoint for your local instance. You need to Check the &ldquo;Searching Using SBOLExplorer&rdquo; to enable the SBOLExplorer. Also, make sure that any of the endpoints you enter are enabled and end with <strong>\</strong>.</p>
<h4 id="79-sparql">7.9 SPARQL</h4>
<p>To perform a SPARQL Search, the following steps should be followed:</p>
<ol>
<li>
<p>Navigate to SynBioHub&rsquo;s <a href="https://synbiohub.org/sparql">SPARQL Search Page</a>.</p>
</li>
<li>
<p>Since, SPARQL is a query language, hence it&rsquo;ll query Synbiohub&rsquo;s databases for the designs that you&rsquo;re looking for.</p>
</li>
<li>
<p>Following is an example how a sample SPARQL query is written:</p>
</li>
</ol>
<pre><code>  SELECT  # makes up the headers of the table
	
        ?def # directs you to the design for which you're looking for 
	
        ?displayId # This statement displays the Id the design
	
        ?title # displays the title of the design
	
        (count(?def) as ?count) # displays the count parameter for the design
	
        ?role # displays the role of the design    

</code></pre><ol start="4">
<li>Next, you need to provide a <strong>Where</strong> statement, which shall contain all the specific details with respect to the <strong>SELECT</strong> statement. The Syntax is as follows:</li>
</ol>
<pre><code>WHERE { 
  	 ?s sbol:component ?comp # Searching for **sbol** components

  	 ?comp sbol:definition ?def # provides the link to that component

         filter (regex(?role, 'http://identifiers.org/so/SO:')) 

	 OPTIONAL {?def sbol:role ?role} # Defines the role

	 OPTIONAL {?def sbol:displayId ?displayId} # gives the ID of the design

	 OPTIONAL {?def dcterms:title ?title} 
  }
</code></pre><p><em>The query selects all sbol components that have an SO role type. It returns the part title, part role, part display id, part uri, and the total number of parts returned by the query. If you want to search for other types of designs, just replace SBOL with your prefered type.</em></p>
<p>To learn more about SPARQL, visit <a href="https://www.w3.org/TR/sparql11-query/">this page</a>.</p>
<h4 id="710-theme">7.10 Theme</h4>
<p>On this several options are available to alter the User interface of your local instance.</p>
<ul>
<li>
<p>You can choose to change the logo of your local instance i.e, replace it with the desalt one bu uploading a picture of your choice. To do so, click on choose file. On doing so, a popup shall appear displaying the .png or the .jpeg files on your computer. Choose the file offs per your convenience and then click on submit.</p>
</li>
<li>
<p>You can also change the name of your Synbiohub local instance, by typing in the name of your choice in the given <strong>instance name</strong> box.</p>
</li>
<li>
<p>There is also an option to change the text that appears on the front page as soon as you open the front page of your local instance. To edit that text, a box is given, wherein the default text is already written. You can simply edit the text, in that very box with the text of your choice.</p>
</li>
<li>
<p>Subsequently, there is a section titled <strong>instance settings</strong>. Under that, you can change the base colour of your Synbiohub instance, which by default hex. To search for more HTML colour codes, click <a href="https://htmlcolorcodes.com/">here</a>.
Also, you can also enable module interactions as well as remove the public enabled.</p>
</li>
</ul>
<p>To save all changes click on <strong>save</strong>.</p>
<h4 id="711-users">7.11 Users</h4>
<p>This page basically lists all the users currently registered onto your local instance.</p>
<ul>
<li>There is an option to add a user for your local instance. To do so, click on <strong>add user</strong>. This&rsquo;ll direct you to page, wherein you&rsquo;ll be required to fill the following details:</li>
</ul>
<ol>
<li>
<p><strong>Full Name</strong>: Enter the name of the person, whose account you want to add.</p>
</li>
<li>
<p><strong>Affiliation</strong>: This is an optional attribute. Enter the university/college, to which that member belongs to.</p>
</li>
<li>
<p><strong>Email</strong>: Enter the Email of the new member.</p>
</li>
<li>
<p><strong>Username</strong>: Assign a username to the member.</p>
</li>
<li>
<p><strong>Grant Administrative Access</strong>: There is an option to provide the administrator access, to this particular member. This shall enable, him/her to edit each and every setting on your local instance including the admin ones as well. To grant this particular access, check the grant administrative privileges box.</p>
</li>
<li>
<p><strong>Grant Member Privileges</strong>: There is an option to provide the member access.
To grant this particular access, check the grant member privileges box.</p>
</li>
<li>
<p><strong>Grant Curator Privileges</strong>: There is an option to provide the member access.
To grant this particular access, check the grant curator privileges box.</p>
</li>
</ol>
<ul>
<li>
<p>You can also search for a particular user by entering the username in the search box.</p>
</li>
<li>
<p>You can choose the number of users, that you want to display on a particular page ranging from 10 to 100.</p>
</li>
<li>
<p>Then, a sortable list of users is displayed. Following is the list of details that is mentioned:</p>
</li>
</ul>
<ol>
<li>
<p><strong>Full Name</strong>: Enter the name of the person, whose account you want to add.</p>
</li>
<li>
<p><strong>Affiliation</strong>: This is an optional attribute. Enter the university/college, to which that member belongs to.</p>
</li>
<li>
<p><strong>Email</strong>: Enter the Email of the new member.</p>
</li>
<li>
<p><strong>Username</strong>: Assign a username to the member.</p>
</li>
<li>
<p><strong>Member</strong>: A checked or an empty box is displayed. A checked box means, that the user has member rights and if not, that means the user doesn&rsquo;t have member access.</p>
</li>
<li>
<p><strong>Curator</strong>: A checked or an empty box is displayed. A checked box means, that the user has curator rights and if not, that means the user doesn&rsquo;t have curator access.</p>
</li>
<li>
<p><strong>Admin</strong> : A checked or an empty box is displayed. A checked box means, that the user has admin rights and if not, that means the user doesn&rsquo;t have admin access.</p>
</li>
<li>
<p><strong>Save</strong>: Clicking on the save option, will permanently save that profile to your local instance.</p>
</li>
</ol>
<p>Finally, there is a check box at the end corresponding to allowing public account creation. Enabling this option will allow anyone to create an account on your local instance and hence contribute to your design repository.</p>


    


                    
                    <div class="row"><div class="position-relative mx-auto col-lg-9">
                          <div class="bg-primary overflow-hidden p-3 mt-5 shadow">

    <h4 class="text-white text-center">Read more</h4>

    <div class="d-flex justify-content-center"><a class="p-1 mr-3 d-inline-block text-white" href="https://synbiohub.github.io/setup/" title="Setup"><i class="fas fa-chevron-left p-1"></i>Setup</a>
        <a class="p-1 ml-3 d-inline-block text-white text-right" href="https://synbiohub.github.io/apidocumentation/" title="API Documentation">API Documentation<i class="fas fa-chevron-right p-1"></i></a>
    </div>
</div>


                        </div></div> 

                </div>

            </div> 

        </div> 
<script src="https://synbiohub.github.iolib/jquery.min.js"></script> 
<script src="https://synbiohub.github.iolib/popper.min.js"></script> 

<script src="https://synbiohub.github.iojs/bootstrap.min.js"></script> 


<script type="text/javascript" src="https://synbiohub.github.io/plugins/lunr.min.js"></script>
<script type="text/javascript" src="https://synbiohub.github.io/plugins/auto-complete.js"></script>
<link href="https://synbiohub.github.io/plugins/auto-complete.css" rel="stylesheet">
<script type="text/javascript">
  
      var baseurl = "https:\/\/synbiohub.github.io";
  
</script>
<script type="text/javascript" src="https://synbiohub.github.io/plugins/search.js"></script>


<script type="text/javascript" src="https://synbiohub.github.io/plugins/clipboard.js"></script>
<script>
  new ClipboardJS('.btn');
</script>
</body>
</html>
